### PR TITLE
Redo build process and hoist compiler parameters to top of file

### LINKED ISSUE #4: Move compiler paths/args to top of file
This change will enable users to easily change the compiler to their preferred one. Currently, the user would have to dig through the codebase to find the part where the compilers are called, which is annoying since the file is so long. Further pushing for this change, the sedimentation assembler bundled with this repository is still in development and very bug prone, so if the user wants a more stable experience, they would most likely want to switch it out for nasm or another equivalent for the time being.

### PR DESCRIPTION
**Pull request by wdotmathree**
Made a unified CMake build system to build the entire project with one command. Also hoisted all the compiler paths and arguments into a section at the top of the file. This aims to fix #4 and make the entire experience more user friendly. Next steps include making port values easier to change and including a script that hosts the entire site in one command as well as centralizing error messages etc. Generally we want to aim to make the project more centralized and not like 3 separate parts.

### CODE REVIEW

**Comment by wdotmathree** [comment]
Very minimal (zero) testing has been done following these changes as there are still some parts of the system that are very difficult to initialize on a new machine. @kevlu8 Please help test these changes.

File: scripts/host.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+(cd ../judging-backend ; node &> /dev/null) &
```
Comment (kevlu8) [inline scripts/host.sh]: I could be wrong, but doesn't `node` just start the node interpreter? This should be `node .`, no?

File: judging-backend/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+enable_language(CXX)
+set(CMAKE_CXX_STANDARD 20)
+enable_language(C)
+set(CMAKE_C_STANDARD 11)
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
```
Comment (kevlu8) [inline judging-backend/CMakeLists.txt:5]: Does this search for `pypy3` or `python3`?

File: scripts/host.sh (hunk 1 above)
Comment (wdotmathree) [reply scripts/host.sh]: `node` should automatically run index.js in the current directory, but we can change it to `node .` just for clarity.

File: judging-backend/CMakeLists.txt (hunk 2 above)
Comment (wdotmathree) [reply judging-backend/CMakeLists.txt:5]: It searches for `python3` by default. I don't know if CMake can search for `pypy3` so I just have an override in the judge file to use `pypy3` instead.

**Comment by wdotmathree** [comment]
oops ignore that until i push the next change

**Comment by wdotmathree** [comment]
@kevlu8 review pls

**Comment by kevlu8** [comment]
I cannot believe you requested a review for one (1) character of code

**Comment by wdotmathree** [comment]
Fuck you I'm merging.

**Comment by kevlu8** [comment]
THATS MY LINE WTF

**Comment by guywhosteals** [comment]
tables-turn

**Pull request merged**